File: android-architecture-component/LatihanViewModel/java/app/src/main/java/com/dicoding/picodiploma/myviewmodel/MainViewModel.java
package com.dicoding.picodiploma.myviewmodel;

import androidx.lifecycle.ViewModel;

// Meng-extend kelas ViewModel ke MainViewModel
public class MainViewModel extends ViewModel {

    // Membuat variable penampung, untuk dipertahankan datanya
    int result = 0;

    // Metode calculate untuk menghitung volume
    void calculate(String width, String height, String length) {

        // Mengubah hasil result dari hasil perkalian lebar, tinggi dan panjang.
        result = Integer.parseInt(width) * Integer.parseInt(height) * Integer.parseInt(length);
    }
}

File: android-architecture-component/LatihanViewModel/kotlin/app/src/main/java/com/dicoding/picodiploma/myviewmodel/MainViewModel.kt
package com.dicoding.picodiploma.myviewmodel

import androidx.lifecycle.ViewModel

// Meng-extend kelas ViewModel ke MainViewModel
class MainViewModel : ViewModel() {
    // Membuat variable penampung, untuk dipertahankan datanya
    var result = 0

    // Metode calculate untuk menghitung volume
    fun calculate(width: String, height: String, length: String) {
        // Mengubah hasil result dari hasil perkalian lebar, tinggi dan panjang.
        result = width.toInt() * height.toInt() * length.toInt()
    }
}